src/components/input/input.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { Control, FieldValues, useController } from 'react-hook-form';
+import { TextStyle } from 'react-native';
+import { TextInput } from 'react-native-paper';
+
+import { AppColor } from 'src/common/enums';
+import styles from './styles';
+
+type IInputProps = {
+  label?: string;
+  placeholder?: string;
+  style?: TextStyle;
+  isRounded?: boolean;
+  name: string;
+  control: Control<FieldValues>;
+};
+
+const Input: React.FC<IInputProps> = ({
+  label,
+  placeholder,
+  style,
+  name,
+  control,
+}) => {
+  const { field } = useController({ name, control });
```

Comment:
I don't think it is a good idea to keep hooks in common components. We might want to use this component without any form-handler.

Only styles should be applied here. The rest we can always pass with existing props